File: create.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX 1024

int main(int argc, char *argv[])
{
  if (argc < 2)
  {
    printf("Veuillez fournir un nom\n");
    return 1;
  }

  char *nom = argv[1];

  char dossier[MAX];
  char commande[MAX];
  char fichier[MAX];
  char routes[MAX];
  char buffer[MAX * 2];

  // -------------------------
  // création dossier
  // -------------------------
  snprintf(dossier, sizeof(dossier),
           "/home/adja/Documents/Aly JS/Projet Gym/script/pages/%s",
           nom);

  snprintf(commande, sizeof(commande),
           "mkdir -p \"%s\"",
           dossier);

  system(commande);

  // -------------------------
  // création fichier JS
  // -------------------------
  snprintf(fichier, sizeof(fichier),
           "/%s.js",
           dossier, nom);

  FILE *fp = fopen(fichier, "w");

  if (!fp)
  {
    printf("Erreur création fichier\n");
    return 1;
  }

  fprintf(fp,
          "import { navigate } from '../router.js';\n\n"
          "const %s = async () => {\n\n"
          "}\n\n"
          "%s.afterRender = () => {\n\n"
          "}\n\n"
          "export default %s;\n",
          nom, nom, nom);

  fclose(fp);

  printf("%s.js créé\n", nom);

  // -------------------------
  // ROUTES (équivalent bash + sed)
  // -------------------------
  snprintf(routes, sizeof(routes),
           "/home/adja/Documents/Aly JS/Projet Gym/script/router.js");

  FILE *file = fopen(routes, "r");

  if (!file)
  {
    printf("Erreur: router.js introuvable\n");
    return 1;
  }

  // lire contenu
  char content[MAX * 10] = "";
  char line[MAX];

  while (fgets(line, sizeof(line), file))
  {
    strcat(content, line);
  }

  fclose(file);

  // vérifier si route existe déjà
  char search[MAX];
  snprintf(search, sizeof(search), "\"/%s\"", nom);

  if (strstr(content, search))
  {
    printf("La route existe déjà\n");
    return 0;
  }

  // construire nouvelle route
  snprintf(buffer, sizeof(buffer),
           "    ,\"/%s\": \"%s\"\n};",
           nom, nom);

  // remplacer le premier "};"
  char *pos = strstr(content, "};");

  if (!pos)
  {
    printf("Erreur: fin de routes introuvable\n");
    return 1;
  }

  // reconstruire fichier
  *pos = '\0';

  strcat(content, buffer);

  // réécrire router.js
  file = fopen(routes, "w");

  if (!file)
  {
    printf("Erreur écriture router.js\n");
    return 1;
  }

  fputs(content, file);
  fclose(file);

  printf("Route ajoutée : /%s\n", nom);

  return 0;
}

File: create.js
const fs = require("fs");
const path = require("path");


const nom = process.argv[2];

// [
//   '/usr/bin/node',           Le programme Node
//   '/home/adja/create.js',    Le fichier exécuté
//   'home'                     Ton premier argument
// ]

// arg = argument
// v = vector (liste/tableau)


if (!nom) {
    console.log("Veuillez fournir un nom");
    process.exit();
}

const chemin = "/home/adja/Documents/Aly JS/Projet Gym/script/pages";

const CIBLE = path.join(chemin, `/${nom}`);

const cheminFichier = path.join(CIBLE, `${nom}.js`);

fs.mkdirSync(CIBLE, { recursive: true });

const contenu =
    `import { navigate } from '../router.js';

const ${nom} = async () => {\n\n}\n

${nom}.afterRender = ()=>{}

export default ${nom}`

fs.writeFileSync(cheminFichier, contenu);

console.log(`${nom}.js créé dans ${CIBLE}`);


const ROUTES = "/home/adja/Documents/Aly JS/Projet Gym/script/router.js";

if (!fs.existsSync(ROUTES)) {
    console.log("Erreur : router.js introuvable");
    process.exit(1);
}

let router = fs.readFileSync(ROUTES, "utf-8");

if (router.includes(`"/${nom}"`)) {
    console.log(`La route /${nom} existe déjà dans router.js`);
    process.exit(0);
}


const newRoute = `    ,"/${nom}": "${nom}"\n`;

const index = router.indexOf("};");

if (index === -1) {
    console.log("Format router.js invalide");
    process.exit(1);
}

const updatedRouter =
    router.slice(0, index) +
    newRoute +
    "};";

fs.writeFileSync(ROUTES, updatedRouter);

console.log(`Route ajoutée : /${nom}`);